### PR TITLE
Separate interface for staging storage

### PR DESCRIPTION
**Pull request by iuwqyir**
### TL;DR

Created a separate interface for staging storage.
Introduced a new `BlockData` structure and refactored the committer and poller to use it, improving data handling efficiency.

### What changed?

- Added a new `BlockData` struct in `block.go` to encapsulate block-related information.
- Refactored the `Commiter` to work with `BlockData` instead of separate block components.
- Updated the `Poller` to use `BlockData` when handling block results.
- Modified storage interfaces to support operations with `BlockData`.
- Simplified data insertion and deletion processes in staging and main storage.

### CODE REVIEW

**Comment by iuwqyir** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/thirdweb-dev/indexer/36?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#36** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/indexer/36?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#33** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/indexer/33?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#32** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/indexer/32?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#31** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/indexer/31?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @iuwqyir and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by AmineAfia — APPROVED** [review]
This is better! I know now from the code if Im reading the part for staging data or main + inserts to staging are much better. I like this!
